packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1273 @@
+    };
+
+    const decodeAgreementDocumentV1FromEvent = (
+      actualAgreementData: StoredEvent,
+      expectedStoreEvent: Omit<StoredEvent, "data" | "event_version">
+    ): AgreementConsumerDocumentAddedV1 => {
+      const actualAgreementDocument:
+        | AgreementConsumerDocumentAddedV1
+        | undefined = protobufDecoder(AgreementConsumerDocumentAddedV1).parse(
+        actualAgreementData.data
+      );
+
+      if (!actualAgreementDocument) {
+        fail(`impossible to decode ${expectedStoreEvent.type} data`);
+      }
+      return actualAgreementDocument;
+    };
+
+    async function getExpectedAgreementEventV1<T>(
+      agreementId: AgreementId | undefined,
+      version: number | undefined,
+      expectedStoreEvent: Omit<StoredEvent, "data" | "event_version">,
+      decoderFunction: (
+        event: StoredEvent,
+        expectedStoreEvent: Omit<StoredEvent, "data" | "event_version">
+      ) => T
+    ): Promise<T> {
+      expect(agreementId).toBeDefined();
+      if (!agreementId) {
+        fail(
+          `Unhandled error: returned agreementId is undefined for ${agreementId}`
+        );
+      }
+
+      const actualAgreementData: StoredEvent | undefined =
+        version !== undefined
+          ? await readAgreementEventByVersion(agreementId, version, postgresDB)
+          : await readLastAgreementEvent(agreementId, postgresDB);
+
+      if (!actualAgreementData) {
+        fail("Creation fails: agreement not found in event-store");
+      }
+
+      expect(actualAgreementData).toMatchObject({
+        ...expectedStoreEvent,
+        event_version: 1,
+      });
+
+      return decoderFunction(actualAgreementData, expectedStoreEvent);
+    }
+
+    function toAgreementStampsV1(agreement: Agreement): StampsV1 {
+      const stamps: StampsV1 = {};
+
+      if (agreement.stamps.submission) {
+        stamps.submission = toStampV1(agreement.stamps.submission);
+      }
+
+      if (agreement.stamps.activation) {
+        stamps.activation = toStampV1(agreement.stamps.activation);
+      }
+
+      if (agreement.stamps.rejection) {
+        stamps.rejection = toStampV1(agreement.stamps.rejection);
+      }
+
+      if (agreement.stamps.suspensionByProducer) {
+        stamps.suspensionByProducer = toStampV1(
+          agreement.stamps.suspensionByProducer
+        );
+      }
+
+      if (agreement.stamps.suspensionByConsumer) {
+        stamps.suspensionByConsumer = toStampV1(
+          agreement.stamps.suspensionByConsumer
+        );
+      }
+
+      if (agreement.stamps.upgrade) {
+        stamps.upgrade = toStampV1(agreement.stamps.upgrade);
+      }
+
+      if (agreement.stamps.archiving) {
+        stamps.archiving = toStampV1(agreement.stamps.archiving);
+      }
+
+      return stamps;
+    }
+
+    async function uploadDocument(path: string, name: string): Promise<void> {
+      await fileManager.storeBytes(
+        config.s3Bucket,
+        config.consumerDocumentsPath,
```

Comment:
Fix here accordingly to previous comment, base path should also include agreement ID

Reply:
in this case it stores consumer document that should be saved in different way instead of agreement, by the way how the descriptor document's path was saved is irrelevant for testing purpose.